proposals/lip-state-model.md

```diff
@@ -0,0 +1,103 @@
+This allows to authenticate the whole state with the tree Merkle root, the state root.
+In this LIP, we define the state model of a Lisk blockchain and the construction of the state tree from which the state root is calculated.
+
+
+## Copyright
+
+This LIP is licensed under the [Creative Commons Zero 1.0 Universal](https://creativecommons.org/publicdomain/zero/1.0/).
+
+
+## Motivation
+
+A blockchain created with the  Lisk SDK is organized in a modular way: each module registered in the chain defines its own state and the possible state transitions, i.e., the transactions defined within the module or the handlers that can be called by other modules.
+This is reflected by the state tree: each module maintains its own key-value storage and sparse Merkle tree, which are finally merged together to form the complete state tree.
+This LIP specifies the general format for the key-value storages and the way in which they are combined together in the state tree. 
+
+Organizing the state of a blockchain as a Merkle tree allows to cryptographically authenticate the whole state with a single hash, the state root.
+The state root property is calculated at the end of the block processing and included in the block header.
+When a certificate is posted to a chain, the state root is used to authenticate the information contained in it against the block header, which has been signed by the sending chain validators.
+For example, a node can verify that the cross-chain messages in the certificate payload have been included in the outbox of the sending chain by recomputing the state root signed in the block header from the outbox root.
+This is done using a "witness", i.e. an array of hashes used to recompute the root starting from the known elements of the tree. 
+
+In the future, the state root could further facilitate the introduction of light clients.
+Light clients would only check the validity of block headers, without fully verifying every transaction in a block.
+The state root can then be used to verify the correctness of a portion of the state, such as a user balance, requested by the light client to a full node. 
+
+Furthermore, the state root could be used for faster synchronization: a node could download a snapshot of the blockchain at a block close to the tip, therefore synchronizing quickly to the last state, while in the background it downloads and verifies blocks in the past. 
+In this case, the state root would be used to verify the validity of the snapshot.
+
+Finally, the state root can be employed to improve the [regenesis protocol](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0035.md). 
+A node could download the state root from a trusted party and then check that the genesis block of a chain defines a state that corresponds to the trusted state root.
+
+
+## Rationale
+
+Each module registered in a chain maintains a separate key-value storage. 
```

Comment:
```suggestion
Each module registered in a chain maintains a separate key-value map. 
```

Reply:
I think there will be one storage interface for all modules so the storage itself is not separated.

Reply:
edited

Reply:
I added an extra sentence to explain that the key-value maps are combined in a single global storage (otherwise the terms storage prefix and storage key seemed a bit odd)

Reply:
Makes sense.